lib/src/metta/runner/mod.rs

```diff
@@ -63,57 +54,69 @@ enum MettaRunnerMode {
 
 pub struct RunnerState<'a> {
     mode: MettaRunnerMode,
+    metta: &'a Metta,
+    parser: Option<SExprParser<'a>>,
+    atoms: Option<&'a [Atom]>,
     interpreter_state: Option<InterpreterState<'a, DynSpace>>,
     results: Vec<Vec<Atom>>,
 }
 
+impl std::fmt::Debug for RunnerState<'_> {
+    fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
+        f.debug_struct("RunnerState")
+            .field("mode", &self.mode)
+            .field("interpreter_state", &self.interpreter_state)
+            .finish()
+    }
+}
+
 impl Metta {
 
     /// A 1-line method to create a fully initialized MeTTa interpreter
     ///
-    /// NOTE: pass `None` for `env_builder` to use the platform environment
-    /// NOTE: This function is appropriate for Rust or C clients, but if other language-specific
-    ///   stdlibs are involved then see the documentation for [Metta::init]
-    pub fn new_rust(env_builder: Option<EnvBuilder>) -> Metta {
+    /// NOTE: pass `None` for `env_builder` to use the common environment
+    pub fn new(env_builder: Option<EnvBuilder>) -> Metta {
+        Self::new_with_stdlib_loader(|_| {}, env_builder)
+    }
+
+    /// Create and initialize a MeTTa interpreter with a language-specific stdlib
+    ///
+    /// NOTE: pass `None` for `env_builder` to use the common environment
+    pub fn new_with_stdlib_loader<F>(loader: F, env_builder: Option<EnvBuilder>) -> Metta
```

Comment:
As we discussed, a custom space argument is needed because the runner may wish to use a different top-level space implementation for &self.  I added back the space parameter in 790ad1e

However, what is the use case for creating a MeTTa with an existing Tokenizer?

Reply:
You are right that after import is added injecting some custom tokenizer doesn't make much sense. Formally it is a part of a parser and  I don't like current way of decoupling this. I believe we can remove a way of injecting tokenizer via constructor safely.